functions/src/src/AddEvent.ts

```diff
@@ -0,0 +1,29 @@
+import * as admin from 'firebase-admin'
+import { getUser } from './GetUser'
+import { UserPermissionLevel } from '../models/UserPermissionLevel'
+import { APIResponse } from '../models/APIResponse'
+import { HouseCompetition } from '../models/HouseCompetition'
+import { Event } from '../models/Event'
+
+export async function addEvent(event: Event): Promise<Boolean> {
+    
+    let user = await getUser(event.creator_id)
+    if (user.permissionLevel === UserPermissionLevel.RESIDENT) {
```

Comment:
You don't have to change it here, but for the future, there is a verifyUserPermissions(User,UserPermissionLevel[]) function that verifies a user belongs to a group of permission levels.